src/mosaic-ns3-server.cc

```diff
@@ -128,6 +142,21 @@ namespace ns3 {
             return 0;
         }
 
+        if (m_commType == CommunicationType::DSRC){
+            if (!m_dsrc_init_complete){
+                m_nodeManager->InitDsrc();
+                m_dsrc_init_complete = true;
+            }
+        }else if (m_commType == CommunicationType::LTE){
```

Comment:
Should we have a check in these blocks to make make sure if we've initialized DSRC that we're at least logging a warning if we get an LTE communication attempt?

Reply:
Given that m_commType is set at the start and doesn't change during operation, adding a warning for LTE attempts when DSRC is initialized currently seems less necessary. However, I'm curious about your perspective:

1. Are there any specific scenarios or future modifications you foresee where m_commType might change during operation?
2. Is there a particular reason or past issue that prompted this suggestion?
3. Should we consider this as a precautionary measure for potential system updates or integrations?

Reply:
I think your logic there sounds well reasoned, I think there's 2 possible things worth considering:
1. I may be misunderstanding the origin of m_commType. I think when typing that comment up, my impression was that the value there came from the messages coming from the MOSAIC Ambassador (e.g. each message from the ambassador had an m_commType value), and thus possible errors on the MOSAIC side could cause us to get mixed values here. In general, if that data comes from an external source, it's worth being defensive against errors in it and just doing a quick check.

2. There is a world in which we might be interested in mixed-protocol simulation, for example possibly studying the interference between DSRC and CV2X. Not high likelihood, but it may be something worth considering.